src/profile-logic/marker-data.js

```diff
@@ -251,22 +252,27 @@ export function getTabFilteredMarkerIndexes(
  * times for the IPC message.
  */
 export class IPCMarkerCorrelations {
-  _correlations: Map<string, IPCSharedData>;
+  _correlations: Map<Tid, Map<number, IPCSharedData>>;
 
   constructor() {
     this._correlations = new Map();
   }
 
-  _makeKey(tid: number, index: number) {
-    return `${tid},${index}`;
-  }
-
-  set(tid: number, index: number, data: IPCSharedData): void {
-    this._correlations.set(this._makeKey(tid, index), data);
+  set(tid: Tid, index: number, data: IPCSharedData): void {
+    let threadData = this._correlations.get(tid);
+    if (!threadData) {
+      threadData = new Map();
+      this._correlations.set(tid, threadData);
+    }
+    threadData.set(index, data);
   }
 
   get(tid: number, index: number): ?IPCSharedData {
```

Comment:
nit: for consistency, let's use Tid instead of number for the first parameter too.